test/CMakeLists.txt

```diff
@@ -228,52 +228,121 @@ set_tests_properties(test_sqlite_perfdb test_perfdb
+    bool_and_f(${MIOPEN_TEST_HALF} ${is_half_check} is_half_check)
+    
+    set(is_bfloat16_check)
+    set(BF16_TEST_DEFAULT FALSE)
+    option_support_check(${PARSE_BF16_ENABLED} ${PARSE_BF16_DISABLED} ${BF16_TEST_DEFAULT} is_bfloat16_check)
+    bool_and_f(${MIOPEN_TEST_BFLOAT16} ${is_bfloat16_check} is_bfloat16_check)
+
+    set(is_int8_check)
+    set(INT8_TEST_DEFAULT FALSE)
+    option_support_check(${PARSE_INT8_ENABLED} ${PARSE_INT8_DISABLED} ${INT8_TEST_DEFAULT} is_int8_check)
+    bool_and_f(${MIOPEN_TEST_INT8} ${is_int8_check} is_int8_check)
+    
+    set(is_miotensile_check)
+    set(MIOTENSILE_TEST_DEFAULT FALSE)
+    option_support_check(${PARSE_MIOTENSILE_ENABLED} ${PARSE_MIOTENSILE_DISABLED} ${MIOTENSILE_TEST_DEFAULT} is_miotensile_check)
+    bool_not_f(${MIOPEN_TEST_MIOTENSILE} NOT_MIOPEN_TEST_MIOTENSILE)
+    bool_or_f(${NOT_MIOPEN_TEST_MIOTENSILE} ${is_miotensile_check} is_miotensile_check)
+
     # Some tests are xDLOPs specific and should not run on gfx900/906 targets.
-    if((NOT (MIOPEN_TEST_INT8 OR MIOPEN_TEST_BFLOAT16 OR MIOPEN_TEST_HALF OR MIOPEN_TEST_MIOTENSILE OR MIOPEN_TEST_NONXDLOPS))
-    OR (MIOPEN_TEST_BFLOAT16 AND ${PARSE_ALLOW_BFLOAT16} AND (NOT MIOPEN_TEST_MIOTENSILE OR ${PARSE_ALLOW_GEMM_FEATURE}))
-    OR (MIOPEN_TEST_HALF AND ${PARSE_ALLOW_HALF} AND (NOT MIOPEN_TEST_MIOTENSILE OR ${PARSE_ALLOW_GEMM_FEATURE}))
-    OR (MIOPEN_TEST_INT8 AND ${PARSE_ALLOW_INT8} AND (NOT MIOPEN_TEST_MIOTENSILE OR ${PARSE_ALLOW_GEMM_FEATURE}))
-    OR (MIOPEN_TEST_NONXDLOPS AND ${PARSE_ALLOW_NONXDLOPS} AND (NOT MIOPEN_TEST_MIOTENSILE OR ${PARSE_ALLOW_GEMM_FEATURE}))
-    OR (MIOPEN_TEST_FLOAT AND (NOT MIOPEN_TEST_MIOTENSILE OR ${PARSE_ALLOW_GEMM_FEATURE})))
-        add_custom_target(${NAME} ${PARSE_UNPARSED_ARGUMENTS})
-        if(NOT PARSE_SKIP_UNLESS_ALL OR MIOPEN_TEST_ALL)
-            add_test(NAME ${NAME} COMMAND ${CMAKE_COMMAND} --build ${CMAKE_CURRENT_BINARY_DIR} --target ${NAME})
+    set(is_gfx906_check)
+    set(GFX906_TEST_DEFAULT TRUE)
+    option_support_check(${PARSE_GFX906_ENABLED} ${PARSE_GFX906_DISABLED} ${GFX906_TEST_DEFAULT} is_gfx906_check)
+    bool_and_f(${MIOPEN_TEST_GFX906} ${is_gfx906_check} is_gfx906_check)
+
+    set(is_gfx908_check)
+    set(GFX908_TEST_DEFAULT FALSE)
+    option_support_check(${PARSE_GFX908_ENABLED} ${PARSE_GFX908_DISABLED} ${GFX908_TEST_DEFAULT} is_gfx908_check)
+    bool_and_f(${MIOPEN_TEST_GFX908} ${is_gfx908_check} is_gfx908_check)
+
+    set(is_full_check)
+    bool_not_f(${PARSE_SKIP_UNLESS_ALL} is_full_check)
+    bool_or_f(${is_full_check} ${MIOPEN_TEST_ALL} is_full_check)
+
+
+    add_custom_target(${NAME} ${PARSE_UNPARSED_ARGUMENTS})
+    add_test(NAME ${NAME} COMMAND ${CMAKE_COMMAND} --build ${CMAKE_CURRENT_BINARY_DIR} --target ${NAME})
+    if(  (is_gfx906_check OR is_gfx908_check)
+     AND (is_miotensile_check AND is_full_check)
+     AND ( is_half_check OR is_bfloat16_check OR is_int8_check OR MIOPEN_TEST_FLOAT)
+    )
+        if(PARSE_PERF_TEST)
```

Comment:
Please replace PERF_TEST with TEST_PERF_DB_RECORD_NOT_FOUND (yes, this looks terrible, so please feel free to invent something better, if you can)))